### PR TITLE
Make CI more reliable

### PR DESCRIPTION
**Pull request by ZzEeKkAa**
Update is targeted to provide more reliable workflow that picks up all the latest dev dependencies:
- Use `conda-forge` instead of `pkgs/main`
- Use same workflow for linux and windows
- Combine multiple github pages into one workflow file
- Get rid of separate actions since they are not making workflow files more reliable
- ~~Docs does not have `numba_dpex` dependency now and that's make doc build much faster~~
- Use mamba as package resolver for environment
- Use `conda mambabuild` instead of `conda build` provided by `boa` for faster conda package build
- Remove redundant cleanup for conda environment (for Windows it was taking more than a minute)
- Improve general pipeline logic to check that conda package has all the required dependencies
- Prevent multiple messages to appear for github documentation preview

PR checks:
- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to an issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: environment/coverage.yml [hunk 1]
```diff
@@ -4,29 +4,23 @@ channels:
...
-    - flake8
-    - black==20.8b1
-    - pytest-cov
-    - pytest-xdist
-    - pexpect
-variables:
-  CHANNELS: -c defaults -c numba -c intel -c numba/label/dev -c dppy/label/dev --override-channels
-  CHANNELS_DEV: -c dppy/label/dev -c defaults -c numba -c intel -c numba/label/dev --override-channels
+  - coverage
+  - pre-commit
```
Comment (diptorupd) [inline environment/coverage.yml:21]: Why do we really need the linter packages here?

Comment (ZzEeKkAa) [reply environment/coverage.yml]: It was just a copy from pip to conda package. Removed. Looks like it is redundant here.

Comment (diptorupd) [reply environment/coverage.yml]: :+1:

File: docs/conf.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 import sphinx_rtd_theme
 
-import numba_dpex
+# import numba_dpex
```
Comment (diptorupd) [inline docs/conf.py:19]: These changes seem to superfluous or at least split out into separate commit.

**Review by diptorupd — APPROVED** [review]
Great set of changes. Thank you very much!
